Sources/Collar/Classes/Collar/LogItem.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  LogItem.swift
+//  Collar
+//
+//  Created by Petar Jadek on 21.10.2025.
+//  Copyright © 2025 Infinum. All rights reserved.
+//
+
+public enum LogType: String, Sendable {
+    case userProperty = "User property"
+    case event = "Event"
+    case screen = "Screen view"
+}
+
+public struct LogItem: CustomStringConvertible, Identifiable, Sendable {
+    public let id = UUID()
+    public let type: LogType
+    public let name: String
+    public let timestamp: Date
+    public let value: String?
+    public let parameters: [String: LoggerJsonValue]?
+
+    private static var dateFormatter: ISO8601DateFormatter {
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+        return formatter
+    }
```

Comment:
This is just a static getter, correct?
It will recreate the dateFormatter on every call?
It's usually recommended to cache the date formatter, since creating a new one can be an expensive operation.
Something like

```suggestion
    private static let dateFormatter: ISO8601DateFormatter =  {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        return formatter
    }()
```